File: 2015/2_1.py
def surface_area(l, w, h):
	return 2*l*w + 2*w*h + 2*h*l

def area(w,h):
	return w*h

file = open("2_1_input.txt", "r")

total_sqft = 0
for dimensions in file:
	l,w,h = dimensions.split("x")
	l = int(l)
	w = int(w)
	h = int(h)
	total_sqft = total_sqft + surface_area(l,w,h) + min(area(l,w), area(w,h), area(l,h))
print total_sqft

File: 2015/2_1.rb
def surface_area(l, w, h)
	return 2*l*w + 2*w*h + 2*h*l
end

def area(w,h)
	return w*h
end

total_sqft = 0
File.open("2_1_input.txt").each_line do |dimensions|
    l,w,h = dimensions.split("x").map(&:to_i)
    total_sqft +=  surface_area(l,w,h) + [area(l,w), area(w,h), area(l,h)].min
end
puts total_sqft